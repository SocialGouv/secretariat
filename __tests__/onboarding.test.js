import { useSession } from "next-auth/react"
import { render } from "@testing-library/react"

jest.mock("next/router", () => ({
  useRouter() {
    return {
      route: "/onboarding",
      pathname: "",
      query: "",
      asPath: "",
    }
  },
}))

beforeAll(() => {
  jest.useFakeTimers()
  jest.setSystemTime(new Date("20 Aug 2020 00:12:00 GMT").getTime())
})

afterAll(() => {
  jest.useRealTimers()
})

it("renders onboarding page", () => {
  useSession.mockImplementation(() => ({
    data: null,
    status: "unauthenticated",
  }))
  // Import has to happen after the jest setup that mocks the Date object
  const Onboarding = require("../src/pages/onboarding").default
  const { container } = render(<Onboarding />)
  expect(container).toMatchSnapshot()
})
