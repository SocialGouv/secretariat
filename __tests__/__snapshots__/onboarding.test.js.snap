// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders onboarding page 1`] = `
<div>
  <main
    class="flex flex-col pt-12"
  >
    <div
      class="relative flex flex-1 justify-center"
    >
      <div
        class="flex-1 flex items-center bg-red-50 rounded"
      >
        <span
          style="box-sizing: border-box; display: inline-block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative; max-width: 100%;"
        >
          <span
            style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%27800%27%20height=%27522%27/%3e"
              style="display: block; max-width: 100%; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px;"
            />
          </span>
          <img
            alt="illustration embarquement"
            data-nimg="intrinsic"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
          <noscript />
        </span>
      </div>
      <div
        class="flex flex-col gap-8 pl-6"
      >
        <div>
          <h2
            class="text-4xl"
          >
            Embarquement 
          </h2>
          <p
            class="text-2xl"
          >
            à la Fabrique Numérique des Ministères Sociaux
          </p>
        </div>
        <div
          class="flex flex-col flex-1 justify-center"
        >
          <div
            class="flex items-center py-8"
          >
            <i
              class="ri-timer-line ri-4x text-blue-france-113-main"
            />
            <p
              class="text-lg ml-6 flex-1"
            >
              Effectuez votre demande d'
              <strong>
                embarquement en quelques instants
              </strong>
               !
            </p>
          </div>
          <hr />
          <div
            class="flex items-center py-8"
          >
            <i
              class="ri-mail-line ri-4x text-blue-france-113-main"
            />
            <p
              class="text-lg ml-6 flex-1"
            >
              Utilisez l'email de vérification pour
               
              <strong>
                confirmer l'authenticité de votre adresse email
              </strong>
              .
            </p>
          </div>
          <hr />
          <div
            class="flex items-center py-8"
          >
            <i
              class="ri-lock-unlock-line ri-4x text-blue-france-113-main"
            />
            <p
              class="text-lg ml-6 flex-1"
            >
              Obtenez vos
               
              <strong>
                accès aux services mis à votre disposition
              </strong>
               par la Fabrique Numérique.
            </p>
          </div>
        </div>
        <div
          class="flex justify-center"
        >
          <a
            class="flex leading-8 btn primary items-center text-3xl py-6 bg-red-marianne-425-main"
            href="/onboarding/request"
          >
            Commencer
            <i
              class="ri-arrow-right-s-line relative left-2"
            />
          </a>
        </div>
      </div>
    </div>
  </main>
</div>
`;
